scripts/Stretch.py

```diff
@@ -0,0 +1,357 @@
+                vPos = np.concatenate([vPos,np.array([[layerSteps[i].x,layerSteps[i].y]])])
+            current_e = layerSteps[i].e
+        if len(vPos):
+            vTrans = np.copy(vPos)
+        if len(vPos)>=2:
+            self.workOnSequence(vPos,vTrans)
+        layergcode = self.generate(layerSteps,iflush,len(layerSteps),vTrans,layergcode)
+        return layergcode
+
+    def dumpPos(self,onestep):
+        sout = ""
+        if onestep.f != self.output_f:
+            self.output_f = onestep.f
+            sout += " F"
+            sout += "{:.0f}".format(self.output_f).rstrip(".")
+        if onestep.x != self.output_x or onestep.y != self.output_y or onestep.z != self.output_z:
+            assert onestep.x >= 0 and onestep.x < 200 # Security
+            self.output_x = onestep.x
+            sout += " X"
+            sout += "{:.3f}".format(self.output_x).rstrip("0").rstrip(".")
+            assert onestep.y >= 0 and onestep.y < 200 # Security
+            self.output_y = onestep.y
+            sout += " Y"
+            sout += "{:.3f}".format(self.output_y).rstrip("0").rstrip(".")
+        if onestep.z != self.output_z:
+            self.output_z = onestep.z
+            sout += " Z"
+            sout += "{:.3f}".format(self.output_z).rstrip("0").rstrip(".")
+        if onestep.e != self.output_e:
+            self.output_e = onestep.e
+            sout += " E"
+            sout += "{:.5f}".format(self.output_e).rstrip("0").rstrip(".")
+        return sout
+
+    def generate(self,layerSteps,i,iend,vPos,layergcode):
+        ipos = 0
+        while i < iend:
+            if layerSteps[i].step == 0:
+                sout = "G0" + self.dumpPos(layerSteps[i])
+                layergcode = layergcode + sout + "\n"
+                ipos = ipos + 1
+            elif layerSteps[i].step == 1:
+                layerSteps[i].x = vPos[ipos][0]
+                layerSteps[i].y = vPos[ipos][1]
+                sout = "G1" + self.dumpPos(layerSteps[i])
+                layergcode = layergcode + sout + "\n"
+                ipos = ipos + 1
+            else:
+                layergcode = layergcode + layerSteps[i].comment + "\n"
+            i = i + 1
+        return layergcode
+
+
+    def workOnSequence(self,vPos,vTrans):
+        if len(vPos)>2 and ((vPos[len(vPos)-1]-vPos[0])**2).sum(0) < 0.3*0.3:
+            self.wideCircle(vPos,vTrans)
+        else:
+            self.wideTurn(vPos,vTrans)
+        self.pushWall(vPos,vTrans)
+        if (len(vPos)):
+            self.vd1 = np.concatenate([self.vd1,np.array(vPos[:-1])])
+            self.vd2 = np.concatenate([self.vd2,np.array(vPos[1:])])
+
+    def wideCircle(self,vPos,vTrans):
+        '''
+        Similar to wideTurn
+        The first and last point of the sequence are the same,
+        so it is possible to extend the end of the sequence with its beginning when seeking for triangles
+        '''
+        d1 = self.line_width / 2.0
```

Comment:
Ok, I made a couple of hours making pylint happy. I hope my code is more readable now :-)